spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/DefaultChannelFactory.java

```diff
@@ -30,21 +29,33 @@
  */
 public class DefaultChannelFactory implements ChannelFactory {
 
-	@Autowired
-	MessageConverterConfigurer messageConverterConfigurer;
+	private final MessageConverterConfigurer messageConverterConfigurer;
+
+	public DefaultChannelFactory(MessageConverterConfigurer messageConverterConfigurer) {
+		this.messageConverterConfigurer = messageConverterConfigurer;
+	}
 
 	@Override
-	public MessageChannel createChannel(String name, Class<?> inputChannelType) throws Exception {
-		MessageChannel messageChannel = createMessageChannel(inputChannelType);
-		messageConverterConfigurer.configureMessageConverters(messageChannel, name);
-		return messageChannel;
+	public PollableChannel createPollableBindableChannel(String name) throws Exception {
+		PollableChannel pollableChannel = new QueueChannel();
+		messageConverterConfigurer.configureMessageConverters(pollableChannel, name);
+		return pollableChannel;
 	}
 
-	private MessageChannel createMessageChannel(Class<?> messageChannelType) {
-		return isPollable(messageChannelType) ? new QueueChannel() : new DirectChannel();
+	@Override
+	public SubscribableChannel createSubscribableBindableChannel(String name) throws Exception {
+		SubscribableChannel subscribableChannel = new DirectChannel();
+		messageConverterConfigurer.configureMessageConverters(subscribableChannel, name);
+		return subscribableChannel;
 	}
 
-	private boolean isPollable(Class<?> channelType) {
-		return PollableChannel.class.equals(channelType);
+	@Override
+	public SubscribableChannel createSubscribableSharedChannel() throws Exception {
```

Comment:
Indeed. Didn't feel like what was happening there warranted the inclusion in the `ChannelFactory` contract.  I was also suggesting renaming the interface, so that the role it plays it's clearer. In a way this *is* a generic factory, but it's only supposed to be used as part of channel binding.